SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -2723,7 +2723,22 @@ void USpatialReceiver::QueueAddComponentOpForAsyncLoad(const Worker_AddComponent
 {
 	EntityWaitingForAsyncLoad& AsyncLoadEntity = EntitiesWaitingForAsyncLoad.FindChecked(Op.entity_id);
 
-	AsyncLoadEntity.PendingOps.AddComponent(Op.entity_id, ComponentData::CreateCopy(Op.data.schema_type, Op.data.component_id));
+	PendingAddComponentWrapper DummyWrapper;
+	DummyWrapper.EntityId = Op.entity_id;
+	DummyWrapper.ComponentId = Op.data.component_id;
+	int32 AddedComponentIndex = AsyncLoadEntity.InitialPendingAddComponents.Find(DummyWrapper);
```

Comment:
rather than creating a dummy, would prefer a find by predicate for the entity ID and component ID